src/System.Private.CoreLib/shared/System/Reflection/LocalVariableInfo.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System.Reflection
+{
+    public class LocalVariableInfo
+    {
+#if CORECLR
+        private RuntimeType _type;
+#endif
+        private bool _isPinned = false;
+        private int _localIndex = 0;
+
+        public virtual Type LocalType
+        { 
+            get 
+            {
+                Debug.Fail("type must be set!");
```

Comment:
Why would we assert on every access? If this is something that needs to be non-null, then `Debug.Assert(_type != null)` seems more appropriate.